Apply post styles into webview

leandroalonso opened the pull request:
Fixes #14210

(Also address #13862, #9525 and #13864)

This PR renders the post content into a webview applying the correct styles.

The styles were extracted from Calypso with a few modifications.

Noticed that header and footer are going to be tackled in the following task.

### To test

(To render the post into a webview remember to enable the feature flag in app settings)

My recommendation to test this PR is to keep one device (or simulator) with the current in-production Reader and the other with this branch. Then, open a post on both and compare — the post should appear **similar** on both. In this phase I'm looking for very outstanding bugs (instead of fixing everything).

Besides that, please:

1. Test with iPhone and iPad (note that in iPad we're using readable margins)
2. Test light and dark mode
3. Test portrait and landscape

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

leandroalonso commented on WordPress/Classes/ViewRelated/Reader/DetailWebView/ReaderWebView.swift:
```diff
@@ -0,0 +1,96 @@
+        <title>Reader Post</title>
+        <meta name='viewport' content='initial-scale=1, maximum-scale=1.0, user-scalable=no'>
+        <link rel="stylesheet" type="text/css"
+              href="https://wordpress.com/calypso/fallback/async-load-blocks-reader-full-post.feb2901cfbd98579fa81.min.css">
```
In the final version, we won't have this link hardcoded here, even because that might change.

We just need to discover a better way to refer to this.

yaelirub commented:
styling for lists and the 3 mentioned issues looks better. 
A few notes:
1. The dividers and quote lines are a bit weak 
![Simulator Screen Shot - iPhone 11 Pro - 2020-05-29 at 13 41 43](https://user-images.githubusercontent.com/1335657/83306783-1533eb80-a1b8-11ea-806b-79c9a96e88fc.png)
![Simulator Screen Shot - iPhone 11 Pro - 2020-05-29 at 13 42 48](https://user-images.githubusercontent.com/1335657/83306793-18c77280-a1b8-11ea-87b4-7e9bbfc956b6.png)

2. When tapping on links it takes some time to load and for a second or two I wasn't sure if the tap was recognized.

3. In some post there seem to be a missing space at the end 
![Simulator Screen Shot - iPhone 8 - 2020-05-29 at 13 47 57](https://user-images.githubusercontent.com/1335657/83306993-7f4c9080-a1b8-11ea-99d8-c0e31b43743c.png)
![Simulator Screen Shot - iPhone 11 Pro - 2020-05-29 at 13 47 20](https://user-images.githubusercontent.com/1335657/83306996-81165400-a1b8-11ea-92f3-9aa6e2a2c65f.png)

*** its worth mentioning that this PR does not handle media, headers, footers or links an @s

leandroalonso commented:
> On iPad it didn't open a webview but used the previous rendering

@yaelirub Did you override the feature flag? (Profile/Me -> App Settings -> Debug -> Toggle on "Reader content displayed in a WebView")

yaelirub commented:
@leandroalonso  shoot I forgot about that 🤦

leandroalonso commented:
@yaelirub about your comments:

> The dividers and quote lines are a bit weak

I changed the divider to always be displayed with 1px height. It was configured to be displayed with 0.5 in retina displays, but I removed that.

The quotes are actually larger than they were before (9px, before it was 6px). Maybe it's the different color that is giving you this impression?

> When tapping on links it takes some time to load and for a second or two I wasn't sure if the tap was recognized.

This will be soon addressed in a separate task!

> In some post there seem to be a missing space at the end

I'll also tackle that on the task that adds the toolbar. Is that ok?

jkmassel commented:
Howdy friends – we're freezing `15.0` today, so I'm bumping this PR forward to `15.1`. If it needs to land in the `15.0` series, please merge it to `release/15.0` and DM me – I'll be happy to create a new beta release!

yaelirub commented:
@leandroalonso , since some issues are going to be handled in following PRs I'm going to approve this one.
